framework/src/modules/chain/synchronizer/synchronizer.js

```diff
@@ -0,0 +1,44 @@
+const BlockSynchronizationMechanism = require('./block_synchronization_mechanism');
+const FastChainSwitchingMechanism = require('./fast_chain_switching_mechanism');
+
+class Synchronizer {
+	constructor() {
+		this.currentStrategy = null;
+		this.blockSynchronizationMechanism = new BlockSynchronizationMechanism();
+		this.fastChainSwitchingMechanism = new FastChainSwitchingMechanism();
```

Comment:
I think, there is no need to initialise both classes in the constructor? Maybe we can do it in  `_determineStrategy` ?

Reply:
My idea was to not to have to initialize the strategy every time `run()` is called, which can be potentially (but hopefully _NOT_) called multiple times

Reply:
If we execute `run` in synchronizer, wouldn't that mean to create mechanism instances with a clear state? I think you are planning to clear the state of the mechanism instances and reuse them. But if the instances are assigned to a variable and not used, gc won't be able to free the memory.

I don't know what you are planning to do there. Just wanted to remind that. I'd create the instance and dispose it in the `run`.

Reply:
That's a good point @yatki, thanks for the input. Unfortunately, at this phase, I am unsure of which state will the instances hold and if it would be needed to be cleared on each run.

I'll keep your point in mind :D